extensions/cli/cli.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"fmt"
+
+	ranchercli "github.com/rancher/shepherd/clients/ranchercli"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	rancher    = "./rancher"
+	context    = "context"
+	projects   = "projects"
+	namespaces = "namespaces"
+	catalog    = "catalog"
+)
+
+// SwitchContext will display the current context and switch to the default one.
+func SwitchContext(client *ranchercli.Client) error {
+	logrus.Infof("Listing the current context...")
+	err := client.Create(rancher, context, "current")
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Switching to the default context...")
+	err = client.Create(rancher, context, "switch")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateAndDeleteProjects will create and delete projects in the local cluster.
+func CreateAndDeleteProjects(client *ranchercli.Client) error {
+	var projectName = namegen.AppendRandomString("projects")
+	var missingCluster = namegen.AppendRandomString("cluster")
+
+	logrus.Infof("Creating a project...")
+	err := client.Create(rancher, projects, "create", "--cluster", "local", projectName)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Validating the project exists...")
+	err = client.Exists(rancher, projects, projectName)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Deleting the project...")
+	err = client.Create(rancher, projects, "delete", projectName)
```

Comment:
Can we add a check here to make sure when we list the projects, the deleted project wont show up?

Reply:
Resolved, spoke offline regarding this.